### ISSUE TITLE
Null reference when lazy loading is used.

### ISSUE THREAD

**Issue by moldziej** [Impetus/kundera#625]
I'm using Kundera 2.13 with Cassandra.

Suppose we have following entity:

``` java
public class Reader {
    @OneToMany(fetch = FetchType.EAGER)
    private Set<Book> books = new HashSet<Book>();
}
```

and for given data in Cassandra there is no Book for particular Reader.

When I'm executing SELECT \* FROM "Reader" LIMIT 100 using EntityManager the result is empty HashSet.

When I change FetchType to LAZY then initially books reference is ProxySet which is fine. But when I access books, e.g. using books.size() then books reference is changed from ProxySet object to null. I think it's a bug. Books should be empty HashSet regardless of whether books was eagerly or lazily loaded.

Reference is set to null in jpa-engine/core/src/main/java/com/impetus/kundera/proxy/collection/AbstractProxyBase.java

``` java
public abstract class AbstractProxyBase implements ProxyCollection
{
    // (...)
    protected void eagerlyLoadDataCollection()
    {
        if (getDataCollection() == null || getDataCollection() instanceof ProxyCollection)
        {
            // (...)
            if (dataCollection instanceof ProxyCollection)
            {
                dataCollection = null;
            }
            PropertyAccessorHelper.set(getOwner(), getRelation().getProperty(), dataCollection);
        }
    }
}
```

I suggest following patch:

``` diff
--- old/jpa-engine/core/src/main/java/com/impetus/kundera/proxy/collection/AbstractProxyBase.java   Fri Jul  4 00:33:42 2014
+++ new/jpa-engine/core/src/main/java/com/impetus/kundera/proxy/collection/AbstractProxyBase.java   Thu Jul 24 13:03:00 2014
@@ -17,6 +17,8 @@

 import java.util.Collection;
 import java.util.Map;
+import java.util.HashSet;
+import java.util.ArrayList;

 import com.impetus.kundera.metadata.KunderaMetadataManager;
 import com.impetus.kundera.metadata.model.EntityMetadata;
@@ -124,7 +126,13 @@

             if (dataCollection instanceof ProxyCollection)
             {
-                dataCollection = null;
+                if (getRelation().getProperty().getType().isAssignableFrom(java.util.Set.class)) {
+                    dataCollection = new HashSet();
+                } else if (getRelation().getProperty().getType().isAssignableFrom(java.util.List.class)) {
+                    dataCollection = new ArrayList();
+                } else {
+                    dataCollection = null;
+                }
             }
             PropertyAccessorHelper.set(getOwner(), getRelation().getProperty(), dataCollection);
         }
```

**Comment by kkmishra** [comment]
Fixed in current trunk. 
@moldziej  thanks for sharing!

**Comment by chhavigangwal** [comment]
Fixed in 2.13.1
